recipes/cassandra-cpp-driver/all/patches/2.15.3/fix-cmake.patch

```diff
@@ -0,0 +1,466 @@
+-  set_target_properties(cassandra PROPERTIES
+-      COMPILE_PDB_NAME "cassandra"
+-      COMPILE_PDB_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
+-  set_target_properties(cassandra PROPERTIES
+-      ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
+-      LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
+-      RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+-  set_target_properties(cassandra PROPERTIES FOLDER "Driver")
+-
+-  target_compile_definitions(cassandra PRIVATE CASS_BUILDING)
+-
+-  if(CASS_USE_BOOST_ATOMIC AND BOOST_LIBRARY_NAME)
+-    add_dependencies(cassandra ${BOOST_LIBRARY_NAME})
+-  endif()
+-  if(LIBUV_LIBRARY_NAME)
+-    add_dependencies(cassandra ${LIBUV_LIBRARY_NAME})
+-  endif()
+-  if(OPENSSL_LIBRARY_NAME)
+-    add_dependencies(cassandra ${OPENSSL_LIBRARY_NAME})
+-  endif()
+-  if(ZLIB_LIBRARY_NAME)
+-    add_dependencies(cassandra ${ZLIB_LIBRARY_NAME})
+-  endif()
++add_library(${PROJECT_LIB_NAME_TARGET}
++  ${SOURCES}
++  $<TARGET_OBJECTS:curl_hostcheck>
++  $<TARGET_OBJECTS:hdr_histogram>
++  $<TARGET_OBJECTS:http-parser>)
++
++if(CASS_USE_ZLIB)
++  target_sources(${PROJECT_LIB_NAME_TARGET} PRIVATE $<TARGET_OBJECTS:minizip>)
+ endif()
+ 
+-if(CASS_BUILD_STATIC)
+-  add_library(cassandra_static STATIC
+-    ${SOURCES}
+-    $<TARGET_OBJECTS:curl_hostcheck_static>
+-    $<TARGET_OBJECTS:hdr_histogram_static>
+-    $<TARGET_OBJECTS:http-parser_static>
+-    $<TARGET_OBJECTS:minizip_static>)
+-  target_link_libraries(cassandra_static ${CASS_LIBS})
+-  target_include_directories(cassandra_static PRIVATE ${INCLUDE_DIRS} ${CASS_INCLUDES})
+-
+-  set_target_properties(cassandra_static PROPERTIES OUTPUT_NAME cassandra_static)
+-  set_target_properties(cassandra_static PROPERTIES VERSION ${PROJECT_VERSION_STRING} SOVERSION ${PROJECT_VERSION_MAJOR})
+-  set_target_properties(cassandra_static PROPERTIES
+-      COMPILE_PDB_NAME "cassandra_static"
+-      COMPILE_PDB_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
+-  set_target_properties(cassandra_static PROPERTIES
+-      ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
+-      LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
+-      RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+-  set_target_properties(cassandra_static PROPERTIES FOLDER "Driver")
+-
+-  target_compile_definitions(cassandra_static PRIVATE CASS_STATIC)
+-
+-  if(CASS_USE_BOOST_ATOMIC AND BOOST_LIBRARY_NAME)
+-    add_dependencies(cassandra_static ${BOOST_LIBRARY_NAME})
+-  endif()
+-  if(LIBUV_LIBRARY_NAME)
+-    add_dependencies(cassandra_static ${LIBUV_LIBRARY_NAME})
+-  endif()
+-  if(OPENSSL_LIBRARY_NAME)
+-    add_dependencies(cassandra_static ${OPENSSL_LIBRARY_NAME})
+-  endif()
+-  if(ZLIB_LIBRARY_NAME)
+-    add_dependencies(cassandra_static ${ZLIB_LIBRARY_NAME})
+-  endif()
++conan_target_link_libraries(${PROJECT_LIB_NAME_TARGET})
++
++if(WIN32)
++ target_link_libraries(${PROJECT_LIB_NAME_TARGET}
++    iphlpapi
```

Comment:
These system libraries should also be added to the conanfile.

Reply:
[Fixed at](https://github.com/conan-io/conan-center-index/pull/3110/commits/c7123d0c8f070a1db034b5e871eac57bf56f4469)